pkg/kernel/mcp/iptables.go

```diff
@@ -0,0 +1,106 @@
+package kernel
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/modelcontextprotocol/go-sdk/mcp"
+	"github.com/ovn-kubernetes/ovn-kubernetes-mcp/pkg/kernel/types"
+)
+
+// GetIptables MCP handler for iptables operations.
+// GetIptables retrieves iptables/ip6tables rules from a Kubernetes node.
+// Automatically detects IPv6 and uses ip6tables when needed.
+func (s *MCPServer) GetIptables(ctx context.Context, req *mcp.CallToolRequest, in types.ListIPTablesParams) (*mcp.CallToolResult, types.Result, error) {
+	iptablesCliAvailable, err := s.UtilityExists(ctx, req, in.Node, in.Image, "iptables")
+	if !iptablesCliAvailable {
+		return nil, types.Result{}, fmt.Errorf("error while getting list of iptables rules: mentioned image does not have iptables utility: %w", err)
+	}
+
+	if err := validateTableName(in.Table); err != nil {
+		return nil, types.Result{}, fmt.Errorf("error while getting list of iptables rules: %w", err)
+	}
+	if err := validateIptablesCommand(in.Command); err != nil {
+		return nil, types.Result{}, fmt.Errorf("error while getting list of iptables rules: %w", err)
+	}
+	if in.FilterParameters != "" {
+		if err := validateParameters(in.FilterParameters); err != nil {
+			return nil, types.Result{}, fmt.Errorf("error while getting list of iptables rules: %w", err)
+		}
+	}
+
+	cmd := newCommand(iptablesCommand(in.FilterParameters))
+	// Defaults to 'filter' table when not specified
+	cmd.addIf(in.Table == "", "-t", "filter")
+	cmd.addIfNotEmpty(in.Table, "-t", in.Table)
+	// Defaults to -L (list) when command not specified
+	cmd.addIf(in.Command == "", "-L")
+	cmd.addIfNotEmpty(in.Command, in.Command)
+	// FilterParameters are invalid with -S command
+	cmd.addIf(in.FilterParameters != "" && in.Command != "-S", strings.Fields(in.FilterParameters)...)
```

Comment:
```suggestion
	// FilterParameters are invalid with -S/--list-rules command
	cmd.addIf(in.FilterParameters != "" && in.Command != "-S" && in.Command != "--list-rules", strings.Fields(in.FilterParameters)...)
```
Both `-S` and `--list-rules` are needed to be checked here.